lib/wavefront-direct-reporter.js

```diff
@@ -128,28 +128,35 @@ class WavefrontDirectReporter extends ScheduledReporter {
   }
 
   appendCounter(counter, points) {
-    let tags = helper.tagger(this.registry.getTags(counter.name), this.globaltags);
-    points.push(formatter.counterPoint(counter, this.prefix, '', tags));
+    const nameAndTags = this.getDecodedKey(counter);
+    points.push(formatter.counterPoint(counter, nameAndTags[0], this.prefix, '', nameAndTags[1]));
   }
 
   appendGauge(gauge, points) {
-    let tags = helper.tagger(this.registry.getTags(gauge.name), this.globaltags);
-    points.push(formatter.gaugePoint(gauge, this.prefix, '', tags));
+    const nameAndTags = this.getDecodedKey(gauge);
+    points.push(formatter.gaugePoint(gauge, nameAndTags[0], this.prefix, '', nameAndTags[1]));
   }
 
   appendMeter(meter, points) {
-    let tags = helper.tagger(this.registry.getTags(meter.name), this.globaltags);
-    points.push(...formatter.meterPoints(meter, this.prefix, '', tags));
+    const nameAndTags = this.getDecodedKey(meter);
+    points.push(...formatter.meterPoints(meter, nameAndTags[0], this.prefix, '', nameAndTags[1]));
   }
 
   appendTimer(timer, points) {
-    let tags = helper.tagger(this.registry.getTags(timer.name), this.globaltags);
-    points.push(...formatter.timerPoints(timer, this.prefix, '', tags));
+    const nameAndTags = this.getDecodedKey(timer);
+    points.push(...formatter.timerPoints(timer, nameAndTags[0], this.prefix, '', nameAndTags[1]));
   }
 
   appendHistogram(histogram, points) {
-    let tags = helper.tagger(this.registry.getTags(histogram.name), this.globaltags);
-    points.push(...formatter.histoPoints(histogram, this.prefix, '', tags));
+    const nameAndTags = this.getDecodedKey(histogram);
+    points.push(...formatter.histoPoints(histogram, nameAndTags[0], this.prefix, '', nameAndTags[1]));
+  }
+
+  getDecodedKey(metric) {
+    const nameAndTags = helper.decodeKey(metric.name)
+    const metricTags = JSON.parse(nameAndTags[1]);
```

Comment:
recommend using array destructuring to make these names more semantically meaningful.